src/core/lib/security/credentials/jwt/json_token.cc

```diff
@@ -158,7 +162,11 @@ void grpc_auth_json_key_destruct(grpc_auth_json_key* json_key) {
     json_key->client_email = nullptr;
   }
   if (json_key->private_key != nullptr) {
+#if OPENSSL_VERSION_NUMBER < 0x30000000L
```

Comment:
In making these changes I was trying to minimize changes that would affect the standard building with BoringSSL, so most everything is guarded with #if OPENSSL_VERSION_NUMBER

I don't think it would be bad to update this if all the APIs are available, but I think those changes could be put off to a different PR, what do you think?

Reply:
Good call, agreed. Let's defer these changes to a follow-up PR.

Reply:
Can we file a bug to track this follow-up?

Please feel free to resolve once this is done.